input/postgres/statements.go

```diff
@@ -253,38 +269,60 @@ func GetStatements(ctx context.Context, server *state.Server, logger *util.Logge
 		}
 
 		if showtext {
-			select {
-			// Since normalizing can take time, explicitly check for cancellations
-			case <-ctx.Done():
-				return nil, nil, nil, ctx.Err()
-			default:
-				fingerprintAndNormalize(key, receivedQuery.String, server, statements, statementTextsByFp)
+			queryKeys = append(queryKeys, key)
+			queryStats = append(queryStats, stats)
+			queryLength = append(queryLength, len(receivedQuery.String))
+			tmpFile.WriteString(receivedQuery.String)
+		} else {
+			if ignoreIOTiming(postgresVersion, "") {
+				stats.BlkReadTime = 0
+				stats.BlkWriteTime = 0
 			}
+			statementStats[key] = stats
 		}
-		if ignoreIOTiming(postgresVersion, receivedQuery) {
-			stats.BlkReadTime = 0
-			stats.BlkWriteTime = 0
-		}
-		statementStats[key] = stats
 	}
 
 	if err = rows.Err(); err != nil {
 		return nil, nil, nil, err
 	}
 
+	tmpFile.Seek(0, io.SeekStart)
```

Comment:
Whilst this doesn't crash when `tmpFile` is not initialized (I don't understand why, but [Go Playground confirms it](tmpFile.Seek(0, io.SeekStart))), it might be best if we wrap this and the for loop below in `if showtext`, just so a future reader doesn't think it might be a problem.

Reply:
Ah, I think that's because the first thing Seek does is call `File.checkValid`, which [returns an error if File is nil](https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/os/file_posix.go;l=235-240). I agree wrapping this in a `showtext` check is worthwhile, and checking the return value from `Seek` may be worth doing, too, to avoid surprises.